includes/class-indieauth-client-discovery.php

```diff
@@ -74,45 +75,63 @@ private function parse( $url ) {
 			return $response;
 		}
 
-		$content = wp_remote_retrieve_body( $response );
-
-		if ( class_exists( 'Masterminds\\HTML5' ) ) {
-			$domdocument = new \Masterminds\HTML5( array( 'disable_html_ns' => true ) );
-			$domdocument = $domdocument->loadHTML( $content );
-		} else {
-			$domdocument = new DOMDocument();
-			libxml_use_internal_errors( true );
-			if ( function_exists( 'mb_convert_encoding' ) ) {
-				$content = mb_convert_encoding( $content, 'HTML-ENTITIES', mb_detect_encoding( $content ) );
+		$content_type = wp_remote_retrieve_header( $response, 'content-type' );
+		if ( 'application/json' === $content_type ) {
+			$json = json_decode( wp_remote_retrieve_body( $response ), true );
+			if ( ! is_array( $json ) && empty( $json ) ) {
+					return;
 			}
-			$domdocument->loadHTML( $content );
-			libxml_use_internal_errors( false );
-		}
-
-		$this->get_mf2( $domdocument, $url );
-		if ( ! empty( $this->mf2 ) ) {
-			if ( array_key_exists( 'name', $this->mf2 ) ) {
-				$this->client_name = $this->mf2['name'][0];
+			if ( ! array_key_exists( 'client_id', $json ) ) {
+				return;
+			}
+			$this->client_id = $json['client_id'];
+			if ( array_key_exists( 'client_name', $json ) ) {
+				$this->client_name = $json['client_name'];
+			}
+			if ( array_key_exists( 'logo_uri', $json ) ) {
+				$this->client_icon = $json['logo_uri'];
 			}
-			if ( array_key_exists( 'logo', $this->mf2 ) ) {
-				if ( is_string( $this->mf2['logo'][0] ) ) {
-					$this->client_icon = $this->mf2['logo'][0];
-				} else {
-					$this->client_icon = $this->mf2['logo'][0]['value'];
+		} elseif ( 'text/html' === $content_type ) {
+			$content = wp_remote_retrieve_body( $response );
+
+			if ( class_exists( 'Masterminds\\HTML5' ) ) {
+				$domdocument = new \Masterminds\HTML5( array( 'disable_html_ns' => true ) );
```

Comment:
Should this library be installed with composer and so be in the `composer.json` file?

Reply:
Removed it. It is an alternative to what we have, but it isn't necessary for what we're doing.